src/configuration.ts

```diff
@@ -189,16 +187,10 @@ export function setMakePath(path: string): void { makePath = path; }
 // Read the path (full or directory only) of the make tool if defined in settings.
 // It represents a default to look for if no other path is already included
 // in "makefile.configurations.makePath".
-function readMakePath(): void {
-    let workspaceConfiguration: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration("makefile");
-    makePath = workspaceConfiguration.get<string>("makePath");
+async function readMakePath(): Promise<void> {
+    makePath = await util.getExpandedSetting<string>("makePath");
     if (!makePath) {
         logger.message("No path to the make tool is defined in the settings file.");
-    } else {
```

Comment:
Why is this part removed? Was it refactored to somewhere else?

Reply:
We call resolvePathToRoot (which needs a different name and will change with a future refactoring... this makes a full path if the given string is rel path and we assume relative to root) after all strings that we read as paths, except makePath. For make, before this PR, we had resolveSubstitutedPath which was only replacing easy case of ${workspaceFolder} or ${workspaceRoot}, but if given a relative path we were not making it full path resolved to root, because if given "make" we want to also search in path. When we see an intentional path (abs or rel) for "make", we don't pull it from the path anymore. Calling resolvePathToRoot on "make" or "make.exe" will not benefit of the functionality of searching for make in path (look at util.ts toolPathInEnv and how we call it).